stl/inc/iterator

```diff
@@ -1802,15 +1830,35 @@ public:
     }
 
 private:
-    _Ptr _Myptr; // underlying pointer
+    _Ptr _Myptr = nullptr; // underlying pointer
 };
 
+_STL_DISABLE_DEPRECATED_WARNING
 template <class _Ptr>
-_NODISCARD unchecked_array_iterator<_Ptr> make_unchecked_array_iterator(const _Ptr _It) noexcept {
+_CXX_DEPRECATE_STDEXT_ARR_ITERS _NODISCARD unchecked_array_iterator<_Ptr> make_unchecked_array_iterator(
+    const _Ptr _It) noexcept {
     return unchecked_array_iterator<_Ptr>(_It);
 }
+_STL_RESTORE_DEPRECATED_WARNING
 _STDEXT_END
 
+#if _HAS_CXX20
+_STD_BEGIN
+_STL_DISABLE_DEPRECATED_WARNING
+template <class _Ty>
+struct _CXX_DEPRECATE_STDEXT_ARR_ITERS pointer_traits<_STDEXT checked_array_iterator<_Ty*>> {

```

Comment:
Deprecating this partial specialization doesn't seem right. The class template and factory function are deprecated, so any attempt to introduce usage will be diagnosed. This partial specialization of `pointer_traits` will typically be used by library code, which is just attempting to handle a `checked_array_iterator` given to it from the user-programmer.